### ISSUE TITLE
Permalinks with uppercase are ignored / not applied

### ISSUE THREAD

**Issue by BenSouchet** [gohugoio/hugo#10095]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.101.0+extended darwin/amd64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?
Yes

### The issue
I created a new Taxonomy (in my config file `.yaml`):
```
taxonomies:
  category: categories
  tag: tags
  projectTag: projectTags
```
and I wanted to customize the permalink (still in the config file):
```
permalinks:
  projectTags: /projects/tag/:slug/
```

### But
The permalink doesn't work, the terms urls are still using default generated permalink (e.g. `/projecttags/test/`)

If I remove the uppercase letter `projectTags` -> `projecttags`, then all is working properly.

**Comment by bep** [comment]
I'm not sure what it says in the documentation, but I would strongly recommend you keep all of this lower cased.

**Comment by BenSouchet** [comment]
Yes, I think I will avoid uppercase letters but I found nothing in the documentation about lowercase for Permalinks or even for lowercase in naming / naming convention for config variables.

The only element that is indicated is when we use function like `isset` we need to provide a full lowercase string with the variable name ([here](https://gohugo.io/functions/isset/#:~:text=All%20site%2Dlevel%20configuration%20keys%20are%20stored%20as%20lower%20case.))

This is still a bug or at least a limitation, and except the custom permalink the usage of a taxonomy with uppercase work well.

**Comment by regisphilibert** [comment]
Out of curiosity, does it work if you create a content file for the taxonomy at `content/projectTags/_index.md`?

**Comment by jmooring** [comment]
Minimal reproducible example:

```bash
git clone --single-branch -b hugo-github-issue-10095 https://github.com/jmooring/hugo-testing hugo-github-issue-10095
cd hugo-github-issue-10095
hugo && tree public
```

config

```
[permalinks]
projectTags = '/projects/tag/:slug/'

[taxonomies]
projectTag = 'projectTags'
```

Expected output:

```text
public/
├── post/
│   ├── test/
│   │   └── index.html
│   └── index.html
├── projects/
│   └── tag/
│       └── foo/
│           └── index.html
├── projecttags/
│   └── index.html   --> this is expected because the permalink setting does not apply to section pages
└── index.html
```

Actual output:

```text
public/
├── post/
│   ├── test/
│   │   └── index.html
│   └── index.html
├── projecttags/
│   ├── foo/
│   │   └── index.html
│   └── index.html
└── index.html
```

**Comment by jmooring** [comment]
This was fixed in v0.123.0.
